src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -445,6 +524,22 @@ private JsonRpcError ReadError(JToken json)
             };
         }
 
+        private void ClearProgressObject(JToken requestId)
+        {
+            long reqId = requestId.Value<long>();
+
+            lock (this.formatterProgressTracker.ProgressLock)
+            {
+                long progressId;
+
+                if (this.formatterProgressTracker.RequestProgressMap.TryGetValue(reqId, out progressId))
+                {
+                    this.formatterProgressTracker.RequestProgressMap.Remove(reqId);
+                    this.formatterProgressTracker.ProgressMap.Remove(progressId);
+                }
+            }
+        }
```

Comment:
This is a lot of accessing of fields directly on a helper class. The fact that we're consuming the other class's lock field directly is a pretty good indication that that class isn't protecting itself enough. Can we move this into the helper class itself?
Also, I don't see similar code in the `MessagePackFormatter`. Was it left out by accident? Once we promote this code into the helper class, let's be sure to call it from the MessagePackFormatter class as well.